SWENT-GRP5/SOLVIT
# Set Providers Info

This PR introduces two new screens for the provider sign-up process, integrates Firebase for storing relevant data, and adds a compact test to simulate provider registration.

**Changes Implemented:**

_First Screen:_

Enables providers to input service-related details:

- Service offered
- Short description
- Minimum price
- Languages spoken
- Profile picture (uploaded from the gallery).

Information is validated and stored in Firebase under the provider's profile.

_Second Screen:_

Proposes the option to create service packages:
Context: As discussed in previous stand-ups, packages are optional as they depend on the seeker's specific needs.
If the provider opts to create packages, they can add up to three packages with:

- Package title
- Price
- Description
- Key features (bullet points).

Packages are stored in a separate Firebase collection with the provider ID as a field for filtering and linkage.

_Firebase Integration:_
Updated Firebase structure:

- Provider profile includes fields for service details.
- A separate collection for packages, linked to the provider via the providerId field for filtering.

**Testing:**

Added a compact test to simulate a complete provider registration:

- Covers filling in service information and (optionally) creating packages.

**Verification Steps:**

Create a provider profile using the new sign-up screens.
Confirm that:

- Provider service details are stored in Firebase under the provider profile.
- Packages (if created) are correctly stored in the packages collection and linked to the provider via providerId.
